src/parser/transform/transform_ddl.cpp

```diff
@@ -26,13 +26,36 @@ std::unique_ptr<Statement> Transformer::transformAlterTable(
     }
 }
 
-std::unique_ptr<Statement> Transformer::transformCreateNodeTable(
-    CypherParser::KU_CreateNodeTableContext& ctx) {
-    auto tableName = transformSchemaName(*ctx.oC_SchemaName());
+std::string Transformer::getPKName(CypherParser::KU_CreateNodeTableContext& ctx) {
+    int pkCount = 0;
     std::string pkName;
+    auto& propertyDefinitions = *ctx.kU_PropertyDefinitions();
+    for (auto& definition : propertyDefinitions.kU_PropertyDefinition()) {
+        if (definition->PRIMARY() && definition->KEY()) {
+            pkCount++;
+            pkName = transformPrimaryKey(*definition->kU_ColumnDefinition());
+        }
+    }
     if (ctx.kU_CreateNodeConstraint()) {
+        pkCount++;
         pkName = transformPrimaryKey(*ctx.kU_CreateNodeConstraint());
     }
+    if (pkCount == 0) {
```

Comment:
Double check if binder already handles this. I vaguely remember there was similar logic in binder.

If so, no need to check here.

Otherwise, change to `throw ParserException("Cannot find primary key.")`

Reply:
Yes, binder currently handle this, but I think this logic should still stay in here because it is more informative. Right now binder's exception returns:
`Error: Binder exception: Primary key  does not match any of the predefined node properties.`
which doesn't really fit with what the problem is, which is that there is no primary key.